run-integration-test/action.yml

```
@@ -27,86 +22,33 @@ outputs:
+
     - name: Extract Test and Instance Configuration
       env:
-        TEST_PARAMETER: ${{ inputs.test-parameter }}
-        TEST_PLATFORM: ${{ inputs.test-platform }}
-        TEST_RUN: ${{ inputs.test-run }}
+        INTERU_VERSION: ${{ inputs.interu-version }}
+        TEST_PROFILE: ${{ inputs.test-profile }}
         GITHUB_DEBUG: ${{ runner.debug }}
       shell: bash
       run: |
         set -euo pipefail
         [ -n "$GITHUB_DEBUG" ] && set -x
 
-        # Create and enter Python virtual env
-        python -m venv .venv
-        . .venv/bin/activate
-
-        # Install proper yq
-        pip install yq==3.4.3
-        yq --version
-
-        #####################################
-        # Extract Kubernetes-related Values #
-        #####################################
-
-        KUBERNETES_DISTRIBUTION=$(echo "$TEST_PLATFORM" | cut -d - -f 1)
-        KUBERNETES_VERSION=$(echo "$TEST_PLATFORM" | cut -d - -f 2)
-        KUBERNETES_ARCHITECTURE=$(echo "$TEST_PLATFORM" | cut -d - -f 3)
-
-        echo "KUBERNETES_DISTRIBUTION=$KUBERNETES_DISTRIBUTION" | tee -a "$GITHUB_ENV"
-        echo "KUBERNETES_VERSION=$KUBERNETES_VERSION" | tee -a "$GITHUB_ENV"
-        echo "KUBERNETES_ARCHITECTURE=$KUBERNETES_ARCHITECTURE" | tee -a "$GITHUB_ENV"
-
-        ##################################
-        # Extract Instance Configuration #
-        ##################################
-
-        INSTANCE_SIZE=$(yq -er '."instance-size"' ./tests/infrastructure.yaml)
-        INSTANCE_TYPE=$(yq -er \
-          --arg kubernetes_distribution "$KUBERNETES_DISTRIBUTION" \
-          --arg kubernetes_architecture "$KUBERNETES_ARCHITECTURE" \
-          --arg instance_size "$INSTANCE_SIZE" \
-          '.[$kubernetes_distribution][$kubernetes_architecture][$instance_size]' \
-          "$GITHUB_ACTION_PATH/instances.yml"
-        )
-
-        # Optional config options
-        CLUSTER_TTL=$(yq -er '."cluster-ttl" // "4h"' ./tests/infrastructure.yaml)
-        INSTANCE_NODES=$(yq -er '.nodes // 1' ./tests/infrastructure.yaml)
-        INSTANCE_DISK=$(yq -er '.disk // 50' ./tests/infrastructure.yaml)
-        TEST_PARALLELISM=$(yq -er '.parallelism // 2' ./tests/infrastructure.yaml)
-
-        echo "INSTANCE_TYPE=$INSTANCE_TYPE" | tee -a "$GITHUB_ENV"
-        echo "CLUSTER_TTL=$CLUSTER_TTL" | tee -a "$GITHUB_ENV"
-        echo "INSTANCE_NODES=$INSTANCE_NODES" | tee -a "$GITHUB_ENV"
-        echo "INSTANCE_DISK=$INSTANCE_DISK" | tee -a "$GITHUB_ENV"
-        echo "TEST_PARALLELISM=$TEST_PARALLELISM" | tee -a "$GITHUB_ENV"
-
-        ############################
-        # Validate Test Parameters #
-        ############################
-
-        if [ -z "$TEST_RUN" ]; then
-          echo "TEST_RUN must be defined and not empty"
-          exit 1
-        fi
+        # Download interu
+        curl -L -o "$TOOL_DIRECTORY/interu" "https://github.com/stackabletech/actions/releases/download/interu-$INTERU_VERSION/interu-x86_64-unknown-linux-gnu"
```

suggestions: Add curl options

- [-s][1]: Silent output to hide the multiple redirects...
- [-S][2]: ... but show errors if there are any.
- [-f][3]: convert most responses above 400 to an error (else the workflow will continue).

```suggestion
        curl -fsSL -o "$TOOL_DIRECTORY/interu" "https://github.com/stackabletech/actions/releases/download/interu-$INTERU_VERSION/interu-x86_64-unknown-linux-gnu"
```

[1]: https://curl.se/docs/manpage.html#-s
[2]: https://curl.se/docs/manpage.html#-S
[3]: https://curl.se/docs/manpage.html#-f